development/workflow.rst

```diff
@@ -0,0 +1,501 @@
+  You don't *need* to be in the repository's directory if you provide the path appropriately.
+- ``-t $USER`` sets the EUPS *tag*.
+  We use this because your username (``$USER``) is an allowed EUPS tag.
+- ``git`` is used as an EUPS *version*.
+  Semantically we default to calling the version "``git``" to indicate this package's version is the HEAD of a git development branch.
+
+In the above ``eups declare`` command we associated the package version "``git``" with the tag "``$USER``."
+Then when we ``setup`` the package, we told EUPS to setup the package and its dependencies with the version associated with the ``$USER`` tag.
+If the ``$USER`` tag isn't found for dependencies, EUPS will revert to using versions of dependencies linked to the ``current`` tag.
+This is why we initially declared the entire lsstsw repository to have the version ``current``.
+
+.. why not setup -j? Means setup *just* this package, no dependencies
+
+.. _lsstsw-workflow-working-ticket-scons:
+
+Step 3. Compile and Test with SCons
+-----------------------------------
+
+Develop the package(s) as you normally would.
+To build the Stack with the newly-developed package, run SCons from the repository of a package being developed:
+
+.. prompt:: bash
+
+   scons -Q -j 6 opt=3 
+
+These flags tell SCons to build with flags:
+
+- ``-Q``: reduce logging to the terminal,
+- ``-j 6``: build in parallel (e.g., with '6' CPUs),
+- ``opt=3``: build with level 3 optimization.
+
+This ``scons`` command will run several targets by default, in sequence:
+
+1. ``lib``: build the C++ code and SWIG interface layer
+2. ``python``: install the Python code
+3. ``tests``: run the test suite
+4. ``example``: compile and run the examples, and
+5. ``doc``: compile Doxygen-based documentation.
+
+You can build a subset of these targets by specifying one explicitly.
+To simply compile C++, SWIG, build the Python package and run tests:
+
+.. prompt:: bash
+
+   scons -q -j 6 opt=3 tests
+
+.. _lsstsw-workflow-jenkins:
+
+Continuous Integration with Jenkins
+===================================
+
+We use a Jenkins instance to run continuous integration tests on the LSST Stack.
+Jenkins tests the Stack against multiple environments, ensuring that your code is robust.
+
+Step 1. Ensure the Code is Pushed
+---------------------------------
+
+``git push`` all commits in development branches of packages to the remote development branches on GitHub.
+
+Step 2. Log into ci.lsst.codes
+------------------------------
+
+Open https://ci.lsst.codes/job/stack-os-matrix/build?delay=0sec in a browser and setup an account if you have not already done so.
+Once logged in you will see the Jenkins job submission page.
+On that page:
+
+1. Enter the name(s) of development branches to include in the build in the **BRANCHES** field.
+2. Click the **Submit** button and wait.
+
+You can monitor builds in the "Bot: Jenkins" HipChat room.
```

Comment:
I think you can use a hipchat: link for people who have the desktop app open.  (It's obtainable from the Web interface, but surprisingly not the app.)